raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -199,18 +216,21 @@ public void initialize(DistributedStateMachine stateMachine) throws IOException
 
         if (quorum.isLeader()) {
             electionTimer.reset(Long.MAX_VALUE);
+            fetchTimer.reset(fetchTimeoutMs);
             onBecomeLeader(quorum.leaderStateOrThrow());
         } else if (quorum.isCandidate()) {
             // If the quorum consists of a single node, we can become leader immediately
             electionTimer.reset(electionTimeoutMs);
+            fetchTimer.reset(Long.MAX_VALUE);
             maybeBecomeLeader(quorum.candidateStateOrThrow());
         } else if (quorum.isFollower()) {
             FollowerState state = quorum.followerStateOrThrow();
             if (quorum.isVoter() && quorum.epoch() == 0) {
                 // If we're initializing for the first time, become a candidate immediately
                 becomeCandidate();
             } else {
-                electionTimer.reset(electionTimeoutMs);
+                fetchTimer.reset(Long.MAX_VALUE);
```

Comment:
If we are a follower, don't we want to set this to `fetchTimeoutMs`?

Reply:
But if there is no leader, then we still want to timeout so that we can become a candidate, right? I think at least one of these timers should always get set.

Reply:
Yes, in the current PR in `becomeUnattachedFollower` I set the `election` timer not the fetch timer, the reason is that when handling vote requests, if the candidate epoch is larger but the end-offset validation failed, we would not grant the vote though would transit to this unattached-follower state, and under this state we would not send fetch to the old leader.

So I think under this case we should be timing on the shorter election timeout to become candidate, and if we later transit to fetching-follower we'd reset timer to fetch timeout.